aspnetcore/blazor/hosting-model-configuration.md

```diff
@@ -0,0 +1,322 @@
+   ```
+
+   Components use the shared *_Layout.cshtml* file for their layout.
+
+1. Add an action to the Home controller:
+
+   ```csharp
+   public IActionResult Blazor()
+   {
+      return View("_Host");
+   }
+   ```
+
+1. Add a low-priority route for the controller action that returns the *_Host.cshtml* view to the endpoint configuration in `Startup.Configure`:
+
+   ```csharp
+   app.UseEndpoints(endpoints =>
+   {
+       ...
+
+       endpoints.MapFallbackToController("Blazor", "Home");
+   });
+   ```
+
+1. Create a *Pages* folder and add routable components to the app. For example:
+
+   ```razor
+   @page "/counter"
+
+   <h1>Counter</h1>
+
+   ...
+   ```
+
+   When using a custom folder to hold the app's components, add the namespace representing the folder to the *Views/_ViewImports.cshtml* file. For more information, see <xref:blazor/components#integrate-components-into-razor-pages-and-mvc-apps>.
+
+### Reflect the connection state in the UI
+
+When the client detects that the connection has been lost, a default UI is displayed to the user while the client attempts to reconnect. If reconnection fails, the user is provided the option to retry.
+
+To customize the UI, define an element with an `id` of `components-reconnect-modal` in the `<body>` of the *_Host.cshtml* Razor page:
+
+```cshtml
+<div id="components-reconnect-modal">
+    ...
+</div>
+```
+
+The following table describes the CSS classes applied to the `components-reconnect-modal` element.
+
+| CSS class                       | Indicates&hellip; |
+| ------------------------------- | ----------------- |
+| `components-reconnect-show`     | A lost connection. The client is attempting to reconnect. Show the modal. |
+| `components-reconnect-hide`     | An active connection is re-established to the server. Hide the modal. |
+| `components-reconnect-failed`   | Reconnection failed, probably due to a network failure. To attempt reconnection, call `window.Blazor.reconnect()`. |
+| `components-reconnect-rejected` | Reconnection rejected. The server was reached but refused the connection, and the user's state on the server is lost. To reload the app, call `location.reload()`. This connection state may result when:<ul><li>A crash in the server-side circuit occurs.</li><li>The client is disconnected long enough for the server to drop the user's state. Instances of the components that the user is interacting with are disposed.</li><li>The server is restarted, or the app's worker process is recycled.</li></ul> |
+
+### Render mode
+
+Blazor Server apps are set up by default to prerender the UI on the server before the client connection to the server is established. This is set up in the *_Host.cshtml* Razor page:
+
+```cshtml
+<body>
+    <app>
+      <component type="typeof(App)" render-mode="ServerPrerendered" />
+    </app>
+
+    <script src="_framework/blazor.server.js"></script>
+</body>
+```
+
+`RenderMode` configures whether the component:
+
+* Is prerendered into the page.
```

Comment:
Do we have an article which actually details out what `prerendering` is?
As a non-native English speaker, it was a bit unclear to me what `prerendering` is. Although I now know it, I think it's still valuable to have a writeup about this - "what is prerendering actually".
Looking at these two sentences

- Is prerendered into the page.
- Is rendered as static HTML on the page


it's not clear with understanding of what prerendering is, what's the actual difference.

Reply:
This topic is the first place where it appears, so I'll define it here before merging.